rs/bitcoin/ckbtc/minter/src/state.rs

```diff
@@ -1285,6 +1301,33 @@ impl IntoIterator for ProcessableUtxos {
     }
 }
 
+impl ProcessableUtxos {
+    pub fn insert_once_suspended_utxo(&mut self, utxo: Utxo, reason: &SuspendedReason) {
+        self.assert_utxo_is_fresh(&utxo);
+        match reason {
+            SuspendedReason::ValueTooSmall => self.previously_ignored_utxos.insert(utxo),
+            SuspendedReason::Quarantined => self.previously_quarantined_utxos.insert(utxo),
+        };
+    }
+
+    pub fn insert_once_new_utxo(&mut self, utxo: Utxo) {
+        self.assert_utxo_is_fresh(&utxo);
+        self.new_utxos.insert(utxo);
+    }
+
+    fn assert_utxo_is_fresh(&self, utxo: &Utxo) {
+        assert!(!self.new_utxos.contains(utxo), "BUG: UTXO is already known");
+        assert!(
+            !self.previously_quarantined_utxos.contains(utxo),
+            "BUG: UTXO is already known"
+        );
+        assert!(
+            !self.previously_ignored_utxos.contains(utxo),
+            "BUG: UTXO is already known"
```

Comment:
nit: maybe add where it is known to the error message so that each assert can give a different error message?

Reply:
done.